swaggerpy/swagger_model.py

```diff
@@ -132,116 +133,80 @@ def cancel(self):
         pass
 
 
-def start_request(http_client, url, headers):
+def start_request(http_client, url, request_options):
     """Download and parse JSON from a URL.
 
     :param http_client: a :class:`swaggerpy.http_client.HttpClient`
     :param url: url for api docs
+    :param request_options: additional fields to send with the request
     :return: an object with a :func`wait` method which returns the api docs
     """
     if is_file_scheme_uri(url):
         return FileEventual(url)
 
-    request_params = {
-        'method': 'GET',
-        'url': url,
-        'headers': headers,
-    }
+    request_params = dict(
+        method='GET',
+        url=url,
+        **request_options)
     return http_client.start_request(request_params)
 
 
-# TODO: remove processors parameter
-class Loader(object):
-    """Abstraction for loading Swagger API's.
-
-    :param http_client: HTTP client interface.
-    :type  http_client: http_client.HttpClient
-    :param processors: List of processors to apply to the API.
-    :type  processors: list of SwaggerProcessor
+def load_resource_listing(
+        url,
+        http_client,
+        base_url=None,
+        request_options=None):
+    """Load a complete swagger api spec and return all schemas compiled
+    into a single dict.
+
+    :param url: url to the swagger spec (file or http)
+    :param http_client: a :class:`swaggerpy.http_client.HttpClient` for
+        performing the requests to fetch api documents.
+    :param base_url: optional url to use as the base url for api doc paths
+    :param request_options: mapping of additional fields to specify in
+        the http request to fetch resources.
```

Comment:
This is now very close to https://github.com/Yelp/swagger_spec_validator/blob/master/swagger_spec_validator/util.py#L12

This supports a few extra parameters, and supports async loading, but otherwise they are functionally pretty similar. They both load and validate the swagger api docs.

One big difference is the return value.  The `swagger_spec_validator` only raises an exception on failure, but does nothing on success. I think we should talk about having the `swagger_spec_validator` actually return a python representation of the api docs.  This functionality is going to be re-usable in both the client (bravado) and the server (pyramid_swagger).